File: files/js/src/main/scala/com/swoval/files/Platform.scala
package com.swoval.files

import scala.scalajs.js
import scala.scalajs.js.annotation.JSImport

object Platform {
  def isMac(): Boolean = os.platform() == "darwin"
  def isWin(): Boolean = os.platform() == "win32"
}
@js.native
@JSImport("os", JSImport.Default)
private object os extends js.Object {
  def platform(): String = js.native
}

File: files/jvm/src/main/java/com/swoval/files/Platform.java
package com.swoval.files;

public class Platform {
  static boolean isMac() {
    return System.getProperty("os.name", "").startsWith("Mac OS X");
  }
  static boolean isWin() {
    return System.getProperty("os.name", "").startsWith("Windows");
  }
}
